app/src/main/java/com/example/pocketcloset/Fragments/CameraFragment.java

```diff
@@ -96,8 +116,65 @@ public void onClick(View v) {
                             }
         });
 
+        ibAdd.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(getContext(), SaveClothingView.class);
+                ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                bitmap.compress(Bitmap.CompressFormat.PNG, 20, stream);
+                byte[] bitmapBytes = stream.toByteArray();
+                ParseFile image = new ParseFile("clothingPhoto", bitmapBytes);
+                try {
+                    image.save();
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+                intent.putExtra(SaveClothingView.EXTRA_CONTACT, image);
+                getContext().startActivity(intent);
+
+                }
+        });
+
+
+    }
+    public File getPhotoFileUri(String fileName) {
+        // Get safe storage directory for photos
+        // Use `getExternalFilesDir` on Context to access package-specific directories.
+        // This way, we don't need to request external read/write runtime permissions.
+        File mediaStorageDir = new File(getContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES), "APP_TAG");
+
+        // Create the storage directory if it does not exist
+        if (!mediaStorageDir.exists() && !mediaStorageDir.mkdirs()){
+            Log.d("APP_TAG", "failed to create directory");
```

Comment:
Similar to the comment above, you'll want to recover from this error. But implementing the error recovery is lower priority than the app's core functionality :) So, consider opening an issue to track this as well.